src/selectors.js

```diff
@@ -32,7 +32,7 @@ const createSelectors = (structure) => {
       }
       lastPathname = pathname
       const match = matchPath(pathname, path)
-      if (!match || !lastMatch || match.url !== lastMatch.url) {
+      if (!match || !lastMatch || match.url !== lastMatch.url || match.isExact !== lastMatch.isExact) {
```

Comment:
Can you add a comment about the reason for comparing isExact here? So, we won't forget in the future.